src/bindings.rs

```diff
@@ -125,6 +127,39 @@ pub(crate) mod ffi {
     }
 }
 
+impl ffi::Size {
+    pub fn new(w: u32, h: u32) -> Self {
+        Self { w, h }
+    }
+}
+impl TryFrom<&ffi::DimensionsRange> for ffi::Size {
+    type Error = PhilipsSlideError;
```

Comment:
here you can create a new type error 
#[derive(Error)]
DimensionsRangeToSizeError{
   StepXNull,
   StepXNull,
   EndXGreaterthanStartX,
   EndYGreaterthanStartY,
}

Then add this error into PhilipsSlideError 
#[transparent]
DimensionsRangeToSize(#[from] DimensionsRangeToSizeError)

Naming is bad but you get the idea haha